notification-portlet-webcomponents/notification-component-icon/src/NotificationIcon.js

```diff
@@ -0,0 +1,150 @@
+import React, {Component} from 'react';
+import {
+  Dropdown,
+  DropdownMenu,
+  DropdownToggle,
+  DropdownItem,
+} from 'reactstrap';
+import FontAwesomeIcon from '@fortawesome/react-fontawesome';
+import {translate} from 'react-i18next';
+
+class NotificationIcon extends Component {
+  state = {
+    isDropdownOpen: false,
+    bearerToken: null,
+    notifications: []
+  };
+
+  fetchBearerToken = async () => {
+    try {
+      // TODO: allow override of hard-coded URI
+      const response = await fetch('/uPortal/api/v5-1/userinfo', {
+        credentials: 'same-origin'
+      });
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      const bearerToken = await response.text();
+      this.setState({bearerToken});
+      setTimeout(function() {
```

Comment:
Special note on an edge case of using setTimeout in React.
The timeout may happen after the component is unregistered which will throw a fatal exception.
React provides a utility to avoid this edge case https://github.com/reactjs/react-timer-mixin

Reply:
That's good to know.  I never would have guessed it.